Scripts/Python/ki/xKIChat.py

```diff
@@ -1208,6 +1217,80 @@ def ChangePassword(self, newPassword):
             return
         PtChangePassword(newPassword)
 
+    ## Overrides the chat area text colors to be a single user-selected color
+    def SetTextColor(self, arg1, arg2=None, arg3=None):
+
+        clamp = lambda val, minVal, maxVal: max(minVal, min(val, maxVal))
+
+        if not arg1:
+            self.chatMgr.AddChatLine(None, PtGetLocalizedString("KI.Errors.MalformedChatSetTextColorCmd"), kChat.SystemMessage)
+            return
+
+        try:
+            colorRed = int(arg1)
+            colorBlue = int(arg2)
+            colorGreen = int(arg3)
+        except ValueError:
+            # arguments weren't valid integer numbers, so we will try treating them as floats below
+            pass
+        else:
+            # args were a valid set of integer numbers
+            colorRed = clamp(colorRed, 0, 255)
+            colorBlue = clamp(colorBlue, 0, 255)
+            colorGreen = clamp(colorGreen, 0, 255)
+            self.chatMgr.chatTextColor = ptColor(colorRed / 255.0, colorBlue / 255.0, colorGreen / 255.0)
+            self.chatMgr.DisplayStatusMessage(PtGetLocalizedString("KI.Chat.TextColorSetValue", [f"({colorRed}, {colorBlue}, {colorGreen})"]))
+            return
+
+        try:
+            colorRed = float(arg1)
+            colorBlue = float(arg2)
+            colorGreen = float(arg3)
+        except ValueError:
+            # arguments weren't valid float numbers, so we will try treating arg1 as a string below
+            pass
+        else:
+            # args were a valid set of float numbers
+            colorRed = clamp(colorRed, 0.0, 1.0)
+            colorBlue = clamp(colorBlue, 0.0, 1.0)
+            colorGreen = clamp(colorGreen, 0.0, 1.0)
+            self.chatMgr.chatTextColor = ptColor(colorRed, colorBlue, colorGreen)
+            self.chatMgr.DisplayStatusMessage(PtGetLocalizedString("KI.Chat.TextColorSetValue", [f"({colorRed}, {colorBlue}, {colorGreen})"]))
+            return
+
+        newColor = arg1.strip().casefold()
+        if newColor in kColorNames:
+            # newColor is a valid localized color name with a corresponding function on the ptColor class
+            colorFn = getattr(ptColor(), kColorNames[newColor])
+            self.chatMgr.chatTextColor = colorFn()
+            self.chatMgr.DisplayStatusMessage(PtGetLocalizedString("KI.Chat.TextColorSetValue", [newColor]))
+        elif newColor == PtGetLocalizedString("KI.Commands.ChatSetTextColorDefaultArg"):
+            # user requested resetting text colors to defaults
+            self.chatMgr.chatTextColor = None
+            self.chatMgr.DisplayStatusMessage(PtGetLocalizedString("KI.Chat.TextColorSetDefault"))
+        else:
+            hexColor = newColor.lstrip("#")
+            if len(hexColor) == 3:
+                # turn RGB into RRGGBB
+                hexColor = f"{hexColor[0]}{hexColor[0]}{hexColor[1]}{hexColor[1]}{hexColor[2]}{hexColor[2]}"
```

Comment:
Am I reading that int triplets are expected to be in hexadecimal format? I would expect to be able to provide them in decimal, eg `/textcolor 255 255 255` because that's how most programs typically deal with int triplets.

Reply:
You are reading it wrong. Look at the first `try-except-else` block for 0-255 int parsing.

Reply:
Ok. Note to self, this is for something like #000.